lib/matplotlib/figure.py

```diff
@@ -1629,10 +1629,10 @@ def _process_layout(layout):
                 for k, v in enumerate(row):
                     if v == empty_sentinel:
                         continue
-                    elif isinstance(v, (int, str, Axes)):
-                        unique_ids.add(v)
-                    else:
+                    elif not isinstance(v, str) and isinstance(v, Iterable):
```

Comment:
hashable?

Reply:
tuples are hashable.

Reply:
so what do you want to do about them? (I don't actually know)

Reply:
Sorry for not communicating well, do we want

```
[['A', (('B', 'C'), ('D', 'D))]]
```

To treat the nested tuple as a single key or as a nested specification?

Reply:
No problem :)
I guess to keep things simple(ish) we should treat any iterable_non_str (inverse of cbook.is_scalar_or_string) as a nested spec?